### PR TITLE
EKF2: move vehicle_odometry publish to method

### PR DESCRIPTION
**Pull request by dagar**
- refactor vehicle_odometry publish to new method
 - only publish vehicle_odometry if using vision

The goal is to minimize the number of publications from EKF2 before we have N of them (https://github.com/PX4/Firmware/pull/14650). This could be gated differently if anyone has an opinion, but I don't see why we should be publishing it by default.

### CODE REVIEW

**Review by bresch — DISMISSED** [review]
Looking at the definition of the `vehicle_odometry` message, this topic is corresponding to the ROS odometry message for aerial vehicles (https://www.ros.org/reps/rep-0147.html#odometry-interface), so I'm not sure if it should only depend on the presence of VIO. Aside from that, I like the refactor.

File: src/modules/ekf2/EKF2.cpp [hunk 1]
```diff
@@ -897,44 +870,14 @@ void EKF2::Run()
...
-
 				// publish vehicle local position data
 				lpos.timestamp = _replay_mode ? now : hrt_absolute_time();
 				_vehicle_local_position_pub.update();
 
-				// publish vehicle odometry data
-				odom.timestamp = _replay_mode ? now : hrt_absolute_time();
-				_vehicle_odometry_pub.publish(odom);
+				// publish vehicle_odometry
+				if (_param_ekf2_aid_mask.get() & (MASK_USE_EVPOS | MASK_USE_EVYAW | MASK_USE_EVVEL)) {
```
Comment (TSC21) [inline src/modules/ekf2/EKF2.cpp:878]: As @bresch said, I don't think this should be limited to VIO, as this data is not specific to vision fusion.

Comment (dagar) [reply src/modules/ekf2/EKF2.cpp:878]: I'm aware, can you suggest an alternative?

Comment (TSC21) [reply src/modules/ekf2/EKF2.cpp:878]: so we want to control the streams right? maybe a parameter? `EKF2_ODOM_PUB`?

Comment (dagar) [reply src/modules/ekf2/EKF2.cpp]: I kept the refactor, we can control it via param or something later.

**Comment by TSC21** [comment]
Thanks @dagar!

**Pull request merged**